src/test/java/edu/ucsb/mapache/controllers/StudentControllerTests.java

```diff
@@ -0,0 +1,330 @@
+package edu.ucsb.mapache.controllers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.View;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.io.IOException;
+import java.io.Reader;
+
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.ucsb.mapache.advice.AuthControllerAdvice;
+import edu.ucsb.mapache.entities.Student;
+import edu.ucsb.mapache.repositories.StudentRepository;
+import edu.ucsb.mapache.services.CSVToObjectService;
+import edu.ucsb.mapache.entities.AppUser;
+@WebMvcTest(value = StudentController.class)
+@WithMockUser
+public class StudentControllerTests {
+  @Autowired
+  private MockMvc mockMvc;
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @Autowired
+  private WebApplicationContext webApplicationContext;
+
+  @MockBean
+  StudentRepository mockStudentRepository;
+  @MockBean
+  AuthControllerAdvice mockAuthControllerAdvice;
+  @MockBean
+  CSVToObjectService mockCSVToObjectService;
+  @MockBean
+  Reader mockReader;
+
+
+  private String userToken() {
+    return "blah";
```

Comment:
maybe we can come up with a better placeholder value than "blah"? might be inappropriate for this context in my opinion

Reply:
What token would I change this to? Considering how other test files like the RoleController uses a similar "blah" user token?